pkg/question/question_test.go

```diff
@@ -1139,3 +1185,400 @@ func cleanupQuestionTest() string {
 	th.RestoreStdin()
 	return th.ReadStdout()
 }
+
+/*
+Question Default Testing
+*/
+
+func TestAskRegion_WithDefault(t *testing.T) {
+	const defaultRegion = "us-west-1"
+	initQuestionTest(t, "\n")
+
+	testEC2.Svc = &th.MockedEC2Svc{
+		Regions: []*ec2.Region{
+			{
+				RegionName: aws.String("us-east-1"),
+			},
+			{
+				RegionName: aws.String("us-east-2"),
+			},
+			{
+				RegionName: aws.String(defaultRegion),
+			},
+			{
+				RegionName: aws.String("us-west-2"),
+			},
+		},
+	}
+
+	answer, err := question.AskRegion(testEC2, defaultRegion)
+	th.Ok(t, err)
+
+	th.Equals(t, defaultRegion, *answer)
+
+	cleanupQuestionTest()
+}
+
+func TestAskLaunchTemplate_WithDefault(t *testing.T) {
+	const defaultTemplateId = "lt-67890"
+	initQuestionTest(t, "\n")
+
+	testEC2.Svc = &th.MockedEC2Svc{
+		LaunchTemplates: []*ec2.LaunchTemplate{
+			{
+				LaunchTemplateId:    aws.String("lt-12345"),
+				LaunchTemplateName:  aws.String("lt-12345"),
+				LatestVersionNumber: aws.Int64(1),
+			},
+			{
+				LaunchTemplateId:    aws.String(defaultTemplateId),
+				LaunchTemplateName:  aws.String(defaultTemplateId),
+				LatestVersionNumber: aws.Int64(1),
+			},
+		},
+	}
+
+	answer := question.AskLaunchTemplate(testEC2, defaultTemplateId)
+	th.Equals(t, defaultTemplateId, *answer)
+
+	cleanupQuestionTest()
+}
+
+func TestAskLaunchTemplateVersion_WithDefault(t *testing.T) {
+	const testTemplateId = "lt-12345"
+	const defaultVersion = 2
+	initQuestionTest(t, "\n")
+
+	testEC2.Svc = &th.MockedEC2Svc{
+		LaunchTemplateVersions: []*ec2.LaunchTemplateVersion{
+			{
+				LaunchTemplateId:   aws.String(testTemplateId),
+				VersionDescription: aws.String("description"),
+				VersionNumber:      aws.Int64(1),
+				DefaultVersion:     aws.Bool(true),
+			},
+			{
+				LaunchTemplateId: aws.String(testTemplateId),
+				VersionNumber:    aws.Int64(defaultVersion),
+				DefaultVersion:   aws.Bool(false),
+			},
+		},
+	}
+
+	answer, err := question.AskLaunchTemplateVersion(testEC2, testTemplateId, strconv.Itoa(defaultVersion))
+	th.Ok(t, err)
+	th.Equals(t, strconv.Itoa(defaultVersion), *answer)
+
+	cleanupQuestionTest()
+}
+
+func TestAskIfEnterInstanceType_WithDefault(t *testing.T) {
+	const defaultInstanceType = "t3.medium"
+	initQuestionTest(t, "\n")
+
+	testEC2.Svc = &th.MockedEC2Svc{
+		InstanceTypes: []*ec2.InstanceTypeInfo{
+			{
+				InstanceType:     aws.String(ec2.InstanceTypeT2Micro),
+				FreeTierEligible: aws.Bool(true),
+			},
+			{
+				InstanceType:     aws.String(defaultInstanceType),
+				FreeTierEligible: aws.Bool(true),
```

Comment:
This test would be more clear if `FreeTierEligible` were `false` for the instance type we want (the default override: t3.medium).

The way `AskIfEnterInstanceType()` selects its initial default (before the override we pass in) is taking the first free-tier-eligible instance from this list. If someone in the future breaks the default override mechanism and swaps the order of this list (making t3.medium the first free-tier-eligible instance and thus the initial default), this test will continue to pass and we won't catch the error.

If instead we set this to `false`, there will be no chance for the test to pass (that is, t3.medium being selected as the final default for the question) _unless_ the default override mechanism is working correctly, which is what we want from a test.

```suggestion
				FreeTierEligible: aws.Bool(false),
```